python/gsw/main.py

```diff
@@ -0,0 +1,370 @@
+
+    sf_report=request.get_json()
+    #index statefield report in elasticsearch
+    sf_res = es.index(index='statefield_report', doc_type='report', body=sf_report)
+    res={"Report Status": sf_res['result']}
+    return res
+
+# Endpoint for getting data from the statefield reports index in elasticsearch
+@app.route("/search-statefields", methods=["GET"])
+@swag_from("endpoint_configs/search_statefields_config.yml")
+def get_statefield():
+    # Connect to elasticsearch
+    es=readIr.es
+
+    imei = int(request.args.get('imei'))
+    statefield = str(request.args.get('statefield'))
+
+    # Get the most recent document in the statefield index which has a given statefield in it
+    search_object={
+        'query': {
+            'exists': {
+                'field': statefield
+            }
+        },
+        "sort": [
+            {
+                "time": {
+                    "order": "desc"
+                }
+            }
+        ],
+        "size": 1
+    }
+    
+    # Get the value of that statefield from the document
+    if es.indices.exists(index='statefield_report_'+str(imei)):
+        res = es.search(index='statefield_report_'+str(imei), body=json.dumps(search_object))
+        if len(res["hits"]["hits"])!=0:
+            most_recent_field=res["hits"]["hits"][0]["_source"][statefield]
+            return most_recent_field
+        else:
+            return "Unable to find field in index: "+"statefield_report_"+str(imei)
+    else:
+        return "Unable to find index: "+"statefield_report_"+str(imei)
+
+# Endpoint for getting data from iridium reports index in elasticsearch. 
+# Will be used to check if we are allowed to send uplinks in radiosession
+@app.route("/search-iridium", methods=["GET"])
+@swag_from("endpoint_configs/search_iridium_config.yml")
+def get_iridium_field():
+    # Connect to elasticsearch
+    es=readIr.es
+
+    #get imei number of the radio that radiosession is connected to
+    imei = int(request.args.get('imei'))
+    field = str(request.args.get('field'))
+    
+    # Get the most recent document in the iridium index which has a given statefield in it
+    search_object={
+        'query': {
+            'exists': {
+                'field': field
+            }
+        },
+        "sort": [
+            {
+                "time": {
+                    "order": "desc"
+                }
+            }
+        ],
+        "size": 1
+    }
+    
+    # Get the value of that statefield from the document
+    if es.indices.exists(index='iridium_report_'+str(imei)):
+        res = es.search(index='iridium_report_'+str(imei), body=json.dumps(search_object))
+        if len(res["hits"]["hits"])!=0:
+            most_recent_field=res["hits"]["hits"][0]["_source"][field]
+            return str(most_recent_field)
+        else:
+            return "Unable to find field in index: "+"iridium_report_"+str(imei)
+    else:
+        return "Unable to find index: "+"iridium_report_"+str(imei)
```

Comment:
Consider using "f" notation for string building https://realpython.com/python-f-strings/